lsp-test/src/Language/LSP/Test.hs

```diff
@@ -981,6 +986,28 @@ resolveCodeLens cl = do
     Right cl -> return cl
     Left error -> throw (UnexpectedResponseError (SomeLspId $ fromJust $ rsp ^. L.id) error)
 
+-- | Returns the inlay hints in the specified range.
+getInlayHints :: TextDocumentIdentifier -> Range -> Session [InlayHint]
+getInlayHints tId range = do
+  rsp <- request SMethod_TextDocumentInlayHint (InlayHintParams Nothing tId range)
+  pure $ absorbNull $ getResponseResult rsp
+
+{- | Returns the inlay hints in the specified range, resolving any with
+ a non empty _data_ field.
+-}
+getAndResolveInlayHints :: TextDocumentIdentifier -> Range -> Session [InlayHint]
+getAndResolveInlayHints tId range = do
+  inlayHints <- getInlayHints tId range
+  for inlayHints $ \inlayHint -> if isJust (inlayHint ^. L.data_) then resolveInlayHint inlayHint else pure inlayHint
```

Comment:
Hmm, this should probably actually check the server capability for resolving inlay hints, same where we do this for other resolution things. I can't remember whether it's easy to get the server capabilities in `lsp-test`, so okay to leave it like this for now unless you feel like fixing it. I'll make an issue.